engine/src/main/scala/cromwell/webservice/MetadataBuilderActor.scala

```diff
@@ -34,7 +34,7 @@ object MetadataBuilderActor {
   private val KeySeparator = ':'
   private val ListNameCaptureID = "listName"
   private val ListIndexCaptureID = "listIndex"
-  private val ListMatcher = "^(.+)\\[([a-zA-Z0-9]+)\\]$".r(ListNameCaptureID, ListIndexCaptureID)
+  private val ListMatcher = "^(.+)\\[([^\\]\\[]+)\\]$".r(ListNameCaptureID, ListIndexCaptureID)
```

Comment:
I ran a basic WDL against JES, so I may have missed something (for example, I didn't scatter). But I didn't see any metadata from the engine nor JES using a `ListIndexCaptureID` except `[0-9]+`.

Unless someone has an existing use case for keeping this more complicated, I suggest replacing `[^\]\[]+` as the capture, and just handling ordinals.

I don't think backend authors sending list metadata like...

key | value
:--- | ---:
backendList[0] | in0
backendList[1] | in1
backendList[2] | in2
... | ...
backendList[99] | in99
backendList[100] | in100
backendList[101] | in101

... expect the json to render the list values via lexicographic ordering...
```json
"backendList": ["in0", "in1", "in10", "in100", "in101", "in11", "in12", "...", "in97", "in98", "in99"]
```


<details>
<summary>metadata from demo run</summary>
<table border="1" style="border-collapse:collapse">
<tr><th>METADATA_JOURNAL_ID</th><th>WORKFLOW_EXECUTION_UUID</th><th>METADATA_KEY</th><th>METADATA_CALL_FQN</th><th>METADATA_CALL_INDEX</th><th>METADATA_CALL_ATTEMPT</th><th>METADATA_VALUE</th><th>METADATA_TIMESTAMP</th></tr>
<tr><td>1</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>id</td><td>NULL</td><td>NULL</td><td>NULL</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>2016-05-25 18:49:58</td></tr>
<tr><td>2</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>status</td><td>NULL</td><td>NULL</td><td>NULL</td><td>Running</td><td>2016-05-25 18:49:59</td></tr>
<tr><td>3</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>start</td><td>NULL</td><td>NULL</td><td>NULL</td><td>2016-05-25T18:49:58.375-04:00</td><td>2016-05-25 18:49:58</td></tr>
<tr><td>4</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>submission</td><td>NULL</td><td>NULL</td><td>NULL</td><td>2016-05-25T18:49:58.375-04:00</td><td>2016-05-25 18:49:58</td></tr>
<tr><td>5</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>backend</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>JES</td><td>2016-05-25 18:50:02</td></tr>
<tr><td>6</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>runtimeAttributes:docker</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>&quot;ubuntu:latest&quot;</td><td>2016-05-25 18:50:02</td></tr>
<tr><td>7</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>start</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T18:50:01.585-04:00</td><td>2016-05-25 18:50:02</td></tr>
<tr><td>8</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionStatus</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>Starting</td><td>2016-05-25 18:50:02</td></tr>
<tr><td>9</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionStatus</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>Running</td><td>2016-05-25 18:50:02</td></tr>
<tr><td>10</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>preemptible</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>false</td><td>2016-05-25 18:50:02</td></tr>
<tr><td>11</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>stderr</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>gs://kshakir-jes-dsde/cromwell-executions/link_files/fe88d4de-5ab0-44c4-95cc-e922057fe163/call-prepare/prepare-stderr.log</td><td>2016-05-25 18:50:02</td></tr>
<tr><td>12</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>stdout</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>gs://kshakir-jes-dsde/cromwell-executions/link_files/fe88d4de-5ab0-44c4-95cc-e922057fe163/call-prepare/prepare-stdout.log</td><td>2016-05-25 18:50:02</td></tr>
<tr><td>13</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>backendLogs:log</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>gs://kshakir-jes-dsde/cromwell-executions/link_files/fe88d4de-5ab0-44c4-95cc-e922057fe163/call-prepare/prepare.log</td><td>2016-05-25 18:50:02</td></tr>
<tr><td>14</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>runtimeAttributes:docker</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>ubuntu:latest</td><td>2016-05-25 18:50:02</td></tr>
<tr><td>15</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>cache:allowResultReuse</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>true</td><td>2016-05-25 18:50:02</td></tr>
<tr><td>16</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>jobId</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>operations/EMeo8dDOKhiJnuSdtLPJkkIgw7vetLsXKg9wcm9kdWN0aW9uUXVldWU</td><td>2016-05-25 18:50:05</td></tr>
<tr><td>17</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>jesOperationStatus</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>Initializing</td><td>2016-05-25 18:50:05</td></tr>
<tr><td>18</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>jesOperationStatus</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>Running</td><td>2016-05-25 18:50:50</td></tr>
<tr><td>19</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>jesOperationStatus</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>Success</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>20</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[0]:description</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>waiting for quota</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>21</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[0]:startTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:50:05Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>22</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[0]:endTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:50:36Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>23</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[1]:description</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>initializing VM</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>24</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[1]:endTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:15.547767898Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>25</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[6]:description</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>delocalizing-files</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>26</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[5]:description</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>running-docker</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>27</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[5]:startTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:25.749764198Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>28</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[5]:endTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:26.140753443Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>29</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[6]:endTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:29.743684971Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>30</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[6]:startTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:26.140753443Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>31</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[7]:description</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>ok</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>32</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[7]:startTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:29.743684971Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>33</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[7]:endTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:29Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>34</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[8]:description</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>cromwell poll interval</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>35</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[8]:startTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:29Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>36</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[8]:endTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:41.115Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>37</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[3]:description</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>pulling-image</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>38</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[2]:endTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:15.54782815Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>39</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[3]:endTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:23.718033158Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>40</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[2]:description</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>start</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>41</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[3]:startTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:15.54782815Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>42</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[4]:endTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:25.749764198Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>43</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[4]:startTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:23.718033158Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>44</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[2]:startTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:51:15.547767898Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>45</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[4]:description</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>localizing-files</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>46</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionEvents[1]:startTime</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T22:50:36Z</td><td>2016-05-25 18:51:41</td></tr>
<tr><td>47</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionStatus</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>Done</td><td>2016-05-25 18:51:42</td></tr>
<tr><td>48</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>end</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>2016-05-25T18:51:41.658-04:00</td><td>2016-05-25 18:51:42</td></tr>
<tr><td>49</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>returnCode</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>0</td><td>2016-05-25 18:51:42</td></tr>
<tr><td>50</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>outputs:out</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>gs://kshakir-jes-dsde/cromwell-executions/link_files/fe88d4de-5ab0-44c4-95cc-e922057fe163/call-prepare/stdout.txt</td><td>2016-05-25 18:51:42</td></tr>
<tr><td>51</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>executionStatus</td><td>link_files.prepare</td><td>NULL</td><td>1</td><td>Done</td><td>2016-05-25 18:51:42</td></tr>
<tr><td>52</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>status</td><td>NULL</td><td>NULL</td><td>NULL</td><td>Succeeded</td><td>2016-05-25 18:51:42</td></tr>
<tr><td>53</td><td>fe88d4de-5ab0-44c4-95cc-e922057fe163</td><td>end</td><td>NULL</td><td>NULL</td><td>NULL</td><td>2016-05-25T18:51:41.704-04:00</td><td>2016-05-25 18:51:42</td></tr></table>
</details>